internal/pkg/manifest/transform.go

```diff
@@ -0,0 +1,208 @@
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "Storage":
+			dstElem := dst.Interface().(*Storage)
+			srcElem := src.Elem().Interface().(Storage)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "Alias":
+			dstElem := dst.Interface().(*Alias)
+			srcElem := src.Elem().Interface().(Alias)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SidecarConfig":
+			dstElem := dst.Interface().(*SidecarConfig)
+			srcElem := src.Elem().Interface().(SidecarConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SubscribeConfig":
+			dstElem := dst.Interface().(*SubscribeConfig)
+			srcElem := src.Elem().Interface().(SubscribeConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "SQSQueue":
+			dstElem := dst.Interface().(*SQSQueue)
+			srcElem := src.Elem().Interface().(SQSQueue)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "EFSConfigOrBool":
+			dstElem := dst.Interface().(*EFSConfigOrBool)
+			srcElem := src.Elem().Interface().(EFSConfigOrBool)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		case "AuthorizationConfig":
+			dstElem := dst.Interface().(*AuthorizationConfig)
+			srcElem := src.Elem().Interface().(AuthorizationConfig)
+			err = mergo.Merge(dstElem, srcElem, mergo.WithOverride, mergo.WithTransformers(workloadTransformer{}))
+		}
+
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func transformPBasic() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		if src.IsNil() {
+			return nil
+		}
+		dst.Set(src)
+		return nil
+	}
+}
+
+// transformImage implements customized merge logic for Image field of manifest.
+// It merges `DockerLabels` and `DependsOn` in the default manager (i.e. with configurations mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+// And then overrides both `Build` and `Location` fields at the same time with the src values, given that they are non-empty themselves.
+func transformImage() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		// Perform default merge
+		dstImage := dst.Interface().(Image)
+		srcImage := src.Interface().(Image)
+
+		err := mergo.Merge(&dstImage, srcImage, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+		if err != nil {
+			return err
+		}
+
+		// Perform customized merge
+		dstBuild := dst.FieldByName("Build")
+		dstLocation := dst.FieldByName("Location")
+
+		srcBuild := src.FieldByName("Build")
+		srcLocation := src.FieldByName("Location")
+
+		if !srcBuild.IsZero() || !srcLocation.IsZero() {
```

Comment:
Good catch!!!!!!!

Reply:
Good catch because:
1. I forgot to use the basic transformer in `transformImage`. The `mergo.merge` call here should have the same flags as the others (so no `overrideWithEmtpyValue`, and use transformer)
2. After removing `overrideWithEmtpyValue` and using transformers, i realized that I'm missing `dst.Set(dstImage)`.

The tests was passing before because of the present of `overrideWithEmtpyValue`. So to answer your question:
> How about the other fields in Image like DependsOn? Does it work when we want to override an empty DependOn using a non-empty DependOn value?

It work for the other fields as well. Just not in the right way.